airbyte-cdk/python/source_declarative_manifest/main.py

```diff
@@ -3,63 +3,47 @@
 #
 
 
-import argparse
 import sys
-from typing import Any, List, Mapping, Tuple
+from typing import Any, List, Mapping
 
 from airbyte_cdk.connector import BaseConnector
 from airbyte_cdk.entrypoint import AirbyteEntrypoint, launch
+from airbyte_cdk.models import AirbyteMessage, ConfiguredAirbyteCatalog
 from airbyte_cdk.sources.declarative.manifest_declarative_source import ManifestDeclarativeSource
+from airbyte_cdk.sources.source import Source
 from connector_builder import connector_builder_handler
 
 
-def create_source(config: Mapping[str, Any]) -> ManifestDeclarativeSource:
+def create_source(config: Mapping[str, Any], debug: bool) -> ManifestDeclarativeSource:
     manifest = config.get("__injected_declarative_manifest")
-    return ManifestDeclarativeSource(manifest)
+    return ManifestDeclarativeSource(manifest, debug)
 
 
-def get_config_from_args(args: List[str]) -> Mapping[str, Any]:
-    command, config_filepath = preparse(args)
-    if command == "spec":
-        raise ValueError("spec command is not supported for injected declarative manifest")
+def handle_connector_request(source: ManifestDeclarativeSource, args: List[str]):
+    # Verify that the correct args are present for the production codepaths.
+    launch(source, args)
 
-    config = BaseConnector.read_config(config_filepath)
 
+def handle_request(args: List[str]):
+    parser = AirbyteEntrypoint.parse_args(args)
+    command = parser.command
+    config_path = parser.config
+    config = BaseConnector.read_config(config_path)
     if "__injected_declarative_manifest" not in config:
         raise ValueError(
             f"Invalid config: `__injected_declarative_manifest` should be provided at the root of the config but config only has keys {list(config.keys())}"
         )
-
-    return config
-
-
-def preparse(args: List[str]) -> Tuple[str, str]:
-    parser = argparse.ArgumentParser()
-    parser.add_argument("command", type=str, help="Airbyte Protocol command")
-    parser.add_argument("--config", type=str, required=True, help="path to the json configuration file")
-    parsed, _ = parser.parse_known_args(args)
-    return parsed.command, parsed.config
-
-
-def handle_connector_builder_request(source: ManifestDeclarativeSource, config: Mapping[str, Any]):
-    command = config.get("__command")
-    if command == "resolve_manifest":
-        return connector_builder_handler.resolve_manifest(source)
-    raise ValueError(f"Unrecognized command {command}.")
-
-
-def handle_connector_request(source: ManifestDeclarativeSource, args: List[str]):
-    # Verify that the correct args are present for the production codepaths.
-    AirbyteEntrypoint.parse_args(args)
-    launch(source, sys.argv[1:])
-
-
-def handle_request(args: List[str]):
-    config = get_config_from_args(args)
-    source = create_source(config)
-    if "__command" in config:
-        print(handle_connector_builder_request(source, config))
+    if command == "read":
```

Comment:
nit: WDYT about putting some exception handling around `get_connector_builder_request_handler` to prevent exceptions from blowing up the production codepath, and moving the `catalog` lines from this function to `get_connector_builder_request_handler` to make a more obvious separation between what is/isn't relevant to a connector builder request.